Flick/Controllers/EditListViewController.swift

```diff
@@ -0,0 +1,272 @@
+        view.addSubview(backButton)
+
+        selectAllButton.setImage(UIImage(named: "selectAll"), for: .normal)
+        selectAllButton.layer.cornerRadius = actionButtonSize.width / 2
+        selectAllButton.addTarget(self, action: #selector(selectAllTapped), for: .touchUpInside)
+        view.addSubview(selectAllButton)
+
+        selectAllLabel.text = "Select all"
+        selectAllLabel.textColor = .white
+        selectAllLabel.font = .systemFont(ofSize: 10, weight: .medium)
+        view.addSubview(selectAllLabel)
+
+        deselectButton.setImage(UIImage(named: "deselect"), for: .normal)
+        deselectButton.layer.cornerRadius = actionButtonSize.width / 2
+        deselectButton.isEnabled = false
+        deselectButton.addTarget(self, action: #selector(deselectTapped), for: .touchUpInside)
+        view.addSubview(deselectButton)
+
+        deselectLabel.text = "Deselect"
+        deselectLabel.textColor = .white
+        deselectLabel.font = .systemFont(ofSize: 10, weight: .medium)
+        view.addSubview(deselectLabel)
+
+        removeButton.setImage(UIImage(named: "remove"), for: .normal)
+        removeButton.layer.cornerRadius = actionButtonSize.width / 2
+        removeButton.isEnabled = false
+        removeButton.addTarget(self, action: #selector(removeTapped), for: .touchUpInside)
+        view.addSubview(removeButton)
+
+        removeLabel.text = "Remove"
+        removeLabel.textColor = .white
+        removeLabel.font = .systemFont(ofSize: 10, weight: .medium)
+        view.addSubview(removeLabel)
+
+        moveButton.setImage(UIImage(named: "move"), for: .normal)
+        moveButton.layer.cornerRadius = actionButtonSize.width / 2
+        moveButton.isEnabled = false
+        moveButton.addTarget(self, action: #selector(moveTapped), for: .touchUpInside)
+        view.addSubview(moveButton)
+
+        moveLabel.text = "Move"
+        moveLabel.textColor = .white
+        moveLabel.font = .systemFont(ofSize: 10, weight: .medium)
+        view.addSubview(moveLabel)
+
+        view.addSubview(roundTopView)
+
+        let mediaCollectionViewLayout = UICollectionViewFlowLayout()
+        mediaCollectionViewLayout.minimumInteritemSpacing = cellPadding
+        mediaCollectionViewLayout.minimumLineSpacing = cellPadding
+        mediaCollectionViewLayout.scrollDirection = .vertical
+
+        mediaCollectionView = UICollectionView(frame: .zero, collectionViewLayout: mediaCollectionViewLayout)
+        mediaCollectionView.backgroundColor = .white
+        mediaCollectionView.register(MediaSelectableCollectionViewCell.self, forCellWithReuseIdentifier: mediaCellReuseIdentifier)
+        mediaCollectionView.dataSource = self
+        mediaCollectionView.delegate = self
+        mediaCollectionView.showsVerticalScrollIndicator = false
+        mediaCollectionView.bounces = false
+        mediaCollectionView.allowsMultipleSelection = true
+        roundTopView.addSubview(mediaCollectionView)
```

Comment:
Do you think its possible to eventually make the media collection view its own component? it seems like its used a lot throughout the app but this is also only helpful if the components are largely the same. if there are too many little differences between then, its just a waste of time to make it its own component haha. just a thought. can make an issue out of this if you think its a good idea...

Reply:
i agree with you. we def have quite many repetitive stuff in our code.